src/bgw/job.c

```diff
@@ -1115,6 +1119,25 @@ ts_job_errors_insert_tuple(const FormData_job_error *job_err)
 	return true;
 }
 
+/**
+ * Die silently without printing a message.
+ *
+ * Also indicate to the postmaster that it does not have to print a message.
+ */
+static void
+choke(SIGNAL_ARGS)
+{
+	/* Flush existing streams so that we can see error message for the rare
+	 * case that proc_exit fails. */
+	fflush(stdout);
```

Comment:
I wonder if it is safe to call `fflush` here?

Reply:
Nope, it's not. It's not async-safe.

Reply:
There are a lot of other async-safe functions in PostgreSQL signal handlers, such as `ereport` and the code above was copied from that function. That is not a reason for us to do that, however, so I replaced it with a plain exist. This will require the jobs using that signal handler to ensure that they do not have any odd behaviors when being terminated, but that should be straightforward.